docs/modules/ROOT/pages/explanation/decisions/portal-api.adoc

```diff
@@ -0,0 +1,78 @@
+
+{product} API only specifies the interface, the implementation is then specific to the provider of the API.
+
+== Relevant requirements
+
+* xref:references/quality-requirements/maintainability/portal-api-modularity.adoc[Portal API Modularity]
+* xref:references/quality-requirements/reliability/portal-api-data-recoverability.adoc[Portal API Data Recoverability]
+* xref:references/quality-requirements/performance/portal-api-data-freshness.adoc[Portal API Data Freshness]
+
+== Proposals
+
+[qanda]
+Custom API implementation::
+Implementation of a custom (REST or GraphQL) API which connects to all the backend system and provides the abstraction to them.
+This can be a synchronous or asynchronous connection, depending on the implementation.
+Authentication, authorization and other core API features will have to be implemented specific to this API.
+
+Kubernetes API::
+Reuse the Kubernetes API to leverage all the core-plumbing which already exists: Authentication, RBAC, scalability, well-defined resource-model, extensibility with `CRDs`, aggregated API server and much more.
+This seems to be a common topic in the Cloud Native world, see for example the experiment called https://github.com/kcp-dev/kcp[kcp - a minimal Kubernetes API server].
+
+== Decision
+
+Kubernetes API
+
+== Rationale
+
+The Kubernetes API server is a perfect match for this use-case. With it, we can many core API plumbing features "for free":
+
+Authentication::
+We can leverage what we already use for the {zone}s.
+Authentication is built-in to the Kubernetes API server and it's even pluggable.
+
+Authorization::
+We can make use of standard Kubernetes RBAC access control and could even enhance it with a policy engine like Kyverno.
+
+Asynchronous::
+It allows us to build upon an asynchronous architecture which when working with multiple backend systems will prove to be more stable and easier to implement.
+
+Persistent Storage::
+Storing data is built-in directly in the core.
+We can store temporary data without having to implement a storage backend.
+
+API Extensibility::
+The API can easily be extended by leveraging `CustomResourceDefinition` (CRDs).
+
+Scalability and Availability::
+It's a core feature of the Kubernetes API server to be highly scalable.
+High-availability possibilities are directly built-in.
+
+API definition vs. Implementation::
+By using `CustomResourceDefinition` (CRDs) to define the API structure, we can easily decouple the interface and the implementation.
+The Portal API client uses the standardized Portal API definition, whereas the implementation in the background communicates with the specific implementation of the backend system.
+
+CLI client::
+We get a CLI client "for free".
+The Kubernetes CLI client `kubectl` can display and work with the custom API endpoints of the Portal API out-of-the-box.
+Think of `kubectl get invoices`.
```

Comment:
These sentences as they are now currently only describe a feature of the proposal, so that should belong to the proposal itself.
Also, there's currently no requirement written down (or mentioned in the Problem section) that would require a CLI in the first place.

Either move these sentences to the proposal, or highlight the advantage that there's no additional effort needed to implement a CLI compared to a REST API (even though a CLI is currently only a future-music feature).

Reply:
Moved to proposal